epilepsy12/tests/view_tests/permissions_tests/test_permissions_delete.py

```diff
@@ -172,35 +174,6 @@ def test_user_delete_success(
             response.status_code == HTTPStatus.OK
         ), f"{test_user.first_name} (from {test_user.organisation_employer}) requested `{url}` for User from {TEST_USER_ORGANISATION}. Has groups: {test_user.groups.all()}. Expected 200 response status code, received {response.status_code}"
 
-        # Additional test for deleting users outside of own Trust
```

Comment:
At first glance this test would still be useful even after this PR?

When I put it back I get an integrity error:

```
FAILED epilepsy12/tests/view_tests/permissions_tests/test_permissions_delete.py::test_user_delete_success - django.db.utils.IntegrityError: duplicate key value violates unique constraint "epilepsy12_epilepsy12user_email_key"
```

Do we know why this fails? I am able to deactivate a user outside of my own organisation in my local instance, although that may be because I'm a super user

Reply:
It fails because after deleting the user, it is reseeding the database with the same user for the next test. Of course in this instance the user has not been deleted, merely deactivated so it throws a database error when trying to create again. I removed the test because I was not clear that it added much, and actually the tests are testing for a 200 or 403 on accessing the function, not actually testing if the user has been removed.

The reason you can edit/update/delete any user is because you have superuser access as you say. If you were to create a lead clinician account and login with that you should be constrained to your own organisation view, and if you try and login and then do a post request through postman (like how many neurologists will do that? but anyway) you should find it throw a 403.

Possibly to close this issue then I might create a test that checks the model rather than the response?

Reply:
[cf655a2](https://github.com/rcpch/rcpch-audit-engine/pull/815/commits/cf655a2e577d3617b5b8622e37f8ca2243c813b1) does this - now deactivation of a user is tested in 2 ways: first that the correct access to the function is given to the correct users, judged by whether a 200 or a 403 is returned; second that any deactivated user is reflected as such in the database.

@mbarton I would be grateful if you are happy with this and the solution, perhaps then merge the request? Thanks again.